core/src/main/java/hudson/util/DescriptorList.java

```diff
@@ -212,7 +212,7 @@ public void load(Class<? extends Describable> c) {
         try {
             Class.forName(c.getName(), true, c.getClassLoader());
         } catch (ClassNotFoundException e) {
-            throw new AssertionError(e);  // Can't happen
+            throw new LinkageError(e.getMessage(), e);  // Can't happen
```

Comment:
`AssertionError` was correct here. Read the above lines carefully.

Reply:
seems to be in the description?

> In addition, I noticed this codebase was using a mixture of Error and AssertionError for "impossible" reflection errors. Error Prone recommends against this in favor of LinkageError for rethrowing ReflectiveOperationException exceptions as unchecked. While the reasoning isn't explicitly stated, presumably it is because LinkageError is more specific. I've updated our usage to match this recommendation.

Reply:
`LinkageError` makes sense if you were actually trying to link something and just happened to be using reflection to do it rather than a static linkage, where the same underlying problem could be thrown in either case with different `Throwable` types: that the actual version of one class is binary incompatible with the expected version. That is not the case here. We expect a `ClassLoader` to be able to `loadClass` a name which we know it already loaded. If that fails, something is wrong with program logic, and it is _not_ a matter of version skew between components. So a `@SuppressWarnings("RethrowReflectiveOperationExceptionAsLinkageError")` would be a more appropriate correction here. Obviously it does not much matter either way.

Reply:
I think we can just go with it, it makes the tool happy and it's not really a problem...

given you approved it I assume you don't feel too strongly on it

Reply:
Agreed.